chirp/drivers/tdh8.py

```diff
@@ -0,0 +1,2479 @@
+STEPS = [2.5, 5.0, 6.25, 10.0, 12.5, 20.0, 25.0, 50.0]
+STEP291_LIST = [str(x) for x in STEPS]
+TDRAB_LIST = ["Off"] + AB_LIST
+TDRCH_LIST = ["CH%s" % x for x in range(1, 129)]
+TIMEOUT_LIST = ["%s sec" % x for x in range(15, 615, 15)] + \
+    ["Off (if supported by radio)"]
+TXPOWER_LIST = ["High", "Low"]
+TXPOWER3_LIST = ["High", "Mid", "Low"]
+VOICE_LIST = ["Off", "English", "Chinese"]
+VOX_LIST = ["OFF"] + ["%s" % x for x in range(1, 11)]
+WORKMODE_LIST = ["Frequency", "Channel"]
+
+SETTING_LISTS = {
+    "almod": ALMOD_LIST,
+    "aniid": PTTID_LIST,
+    "displayab": AB_LIST,
+    "dtmfst": DTMFST_LIST,
+    "dtmfspeed": DTMFSPEED_LIST,
+    "mdfa": MODE_LIST,
+    "mdfb": MODE_LIST,
+    "ponmsg": PONMSG_LIST,
+    "pttid": PTTID_LIST,
+    "rtone": RTONE_LIST,
+    "rogerrx": ROGERRX_LIST,
+    "rpste": RPSTE_LIST,
+    "rxled": COLOR_LIST,
+    "save": SAVE_LIST,
+    "scode": PTTIDCODE_LIST,
+    "screv": RESUME_LIST,
+    "sftd": SHIFTD_LIST,
+    "stedelay": STEDELAY_LIST,
+    "step": STEP_LIST,
+    "step291": STEP291_LIST,
+    "tdrab": TDRAB_LIST,
+    "tdrch": TDRCH_LIST,
+    "timeout": TIMEOUT_LIST,
+    "txled": COLOR_LIST,
+    "txpower": TXPOWER_LIST,
+    "txpower3": TXPOWER3_LIST,
+    "voice": VOICE_LIST,
+    "vox": VOX_LIST,
+    "widenarr": BANDWIDTH_LIST,
+    "workmode": WORKMODE_LIST,
+    "wtled": COLOR_LIST
+}
+
+GMRS_FREQS1 = [462.5625, 462.5875, 462.6125, 462.6375, 462.6625,
+               462.6875, 462.7125]
+GMRS_FREQS2 = [467.5625, 467.5875, 467.6125, 467.6375, 467.6625,
+               467.6875, 467.7125]
+GMRS_FREQS3 = [462.5500, 462.5750, 462.6000, 462.6250, 462.6500,
+               462.6750, 462.7000, 462.7250]
+GMRS_FREQS_B = GMRS_FREQS2 * 3 + GMRS_FREQS3 * 2
+
+GMRS_FREQ_A = GMRS_FREQS1 + GMRS_FREQS3
+
+HAM_FREQS = [162.55000, 162.40000, 162.47500, 162.42500, 162.45000,
+             162.50000, 162.52500, 161.65000, 161.77500, 161.75000]
+HAM_NAME = ["NOAA 1", "NOAA 2", "NOAA 3", "NOAA 4", "NOAA 5", "NOAA 6",
+            "NOAA 7", "NOAA 8", "NOAA 9", "NOAA 10", "NOAA 11"]
+
+ALL_MODEL = ["TD-H8", "TD-H8-HAM", "TD-H8-GMRS"]
+
+
+def _do_status(radio, block):
+    status = chirp_common.Status()
+    status.msg = "Cloning"
+    status.cur = block
+    status.max = radio._memsize
+    radio.status_fn(status)
+
+
+TD_H8 = b"\x50\x56\x4F\x4A\x48\x1C\x14"
+
+
+def _upper_band_from_data(data):
+    return data[0x03:0x04]
+
+
+def _upper_band_from_image(radio):
+    return _upper_band_from_data(radio.get_mmap())
+
+
+def _firmware_version_from_data(data, version_start, version_stop):
+    version_tag = data[version_start:version_stop]
+    return version_tag
+
+
+def _firmware_version_from_image(radio):
+    version = _firmware_version_from_data(radio.get_mmap(),
+                                          radio._fw_ver_file_start,
+                                          radio._fw_ver_file_stop)
+    # LOG.debug("_firmware_version_from_image: " + util.hexprint(version))
```

Comment:
I mentioned this before, but there is a lot of commented-out code in this file. Please remove it all, it makes it very hard to review.